lib/validate.js

```
@@ -0,0 +1,14 @@
+/**
+ * Verify environment:
+ * - Runs on Travis
+ * - Has a Github authentication token
+ * - Runs on after_success step (TRAVIS_TEST_RESULT is set)
+ * 
+ * @param {Object} env Environment variables
+ * @throws {Error} if one of the verification fails
+ */
+module.exports = env => {
+  if (env.TRAVIS !== 'true') throw new Error('Not running on Travis');
+  if (!env.GH_TOKEN) throw new Error('GitHub authentication missing');
+  if (!env.TRAVIS_TEST_RESULT) throw new Error('Not running in Travis after_success step');
```

Are you sure `TRAVIS_TEST_RESULT` cannot be falsy? Maybe we should check with `!('TRAVIS_TEST_RESULT' in env)`?

---

`TRAVIS_TEST_RESULT` is set to `'0'` or `'1'` after the test and in `after_success`.
So we want to handle all the cases in which `TRAVIS_TEST_RESULT` has not been set. To be safe we should handle the case of `undefined`, `null`, `false`.

For info, currently Travis doesn't define the variable until `after_success`, so in JS `process.env.TRAVIS_TEST_RESULT` is `undefined` in steps before `after_success`.
But if for some reasons Travis start to set it as `false` or `null` in steps before `after_success` we still want to throw the Error here, because according to the Travis doc set means `'0'` or `'1'`, unset is not specified.

See https://travis-ci.org/vanduynslagerp/test-travis/builds/279243471#L485

---

I just want to make sure that the `'0'` string does not get parsed to `0` which would make the `!env.TRAVIS_TEST_RESULT` return true

---

`!('TRAVIS_TEST_RESULT' in env)` wouldn't work because if `env` has a `TRAVIS_TEST_RESULT` property that  is set to `undefined` then `!('TRAVIS_TEST_RESULT' in env)` returns `false`.

`'0'` wouldn't get parsed to `0` but to be safe the safest would be `if (typeof env.TRAVIS_TEST_RESULT === 'undefined' || env.TRAVIS_TEST_RESULT === false || env.TRAVIS_TEST_RESULT === null)`.

---

I just made that change and added some more unit test to cover all possible cases